Bug: `#[use_inject]` macro generates non-existent `InjectionContext::new()` in fallback path

## Bug Report

### Description

The `#[use_inject]` procedural macro generates `InjectionContext::new()` in the fallback path when no DI context is set on the router. However, `InjectionContext` does not have a `new()` method — it only supports construction via the builder pattern (`InjectionContext::builder(singleton_scope).build()`).

This causes compilation failures when building test targets (`cargo check --tests`) or any code that uses `#[use_inject]` with `#[inject]` parameters.

### Steps to Reproduce

```bash
cargo check --workspace --all --all-features --tests
```

### Expected Behavior

The macro should generate valid Rust code using the builder pattern for `InjectionContext` construction.

### Actual Behavior

Compilation fails with:
```
error[E0599]: no function or associated item named `new` found for struct `InjectionContext` in the current scope
```

9 errors in `tests/integration/tests/di/macros_integration.rs` at every `#[use_inject]` usage.

### Root Cause

In `crates/reinhardt-core/macros/src/use_inject.rs` line 209, the macro generates:
```rust
Arc::new(InjectionContext::new())
```

Additionally, there is a type mismatch: `get_di_context::<Arc<InjectionContext>>()` returns `Option<Arc<Arc<InjectionContext>>>` (because `set_di_context` wraps the value in `Arc`), so the `None` branch must also return `Arc<Arc<InjectionContext>>`.

### Environment

- Rust: 1.91.1
- OS: macOS / Linux (CI)
- Affected CI checks: `Cargo Check (tests)`, `Cross-Crate Integration Coverage`

--- a/crates/reinhardt-core/macros/src/use_inject.rs
+++ b/crates/reinhardt-core/macros/src/use_inject.rs
@@ -206,7 +206,9 @@ pub(crate) fn use_inject_impl(_args: TokenStream, input: ItemFn) -> Result<Token
 						"DI context not set on router. Creating empty fallback context. \
 						 Hint: Configure the router with .with_di_context() for proper dependency injection."
 					);
-					::std::sync::Arc::new(#di_crate::InjectionContext::new())
+					::std::sync::Arc::new(::std::sync::Arc::new(
+						#di_crate::InjectionContext::builder(#di_crate::SingletonScope::new()).build()
+					))
 				}
 			};
 		}

